src/utils/configOverride.ts

```diff
@@ -1,16 +1,31 @@
 import type { NavigateFunction } from 'react-router/dist/lib/hooks';
 
-import { logDev } from '#src/utils/common';
+import { IS_PROD_MODE, logDev } from '#src/utils/common';
 import type { Settings } from '#src/stores/SettingsStore';
 
 // Use local storage so the override persists until cleared
 const storage = window.localStorage;
 
 const configQueryKey = 'app-config';
 const configLegacyQueryKey = 'c';
+const envQueryKey = 'env';
 
 const configFileStorageKey = 'config-file-override';
 
+export const getApiEnv = (searchParams: URLSearchParams): 'dev' | 'prd' => {
+  const query = searchParams.get(envQueryKey);
+
+  if (IS_PROD_MODE) {
```

Comment:
I'm not sure this will result in static enough code for the compiler to optimize it through method calls and remove all the unused code in prod mode. Maybe something closer to the usages like a `getHost` method that returns the hardcoded value in prod or the dynamic value otherwise?